### PR TITLE
Feature/chat messages

### PR DESCRIPTION
**Pull request by KovalevAnton**
add uploading pictures to chat messages

### CODE REVIEW

File: src/actions/chat.ts [hunk 1]
```diff
@@ -215,3 +218,53 @@ export const refreshChatList = () => async (dispatch) => {
...
+      dispatch({
+        type: UPLOAD_PROGRESS,
+        payload: written / total
+      });
+    })
+    .progress((received, total) => {
+      dispatch({
+        type: UPLOAD_END
+      });
+      dispatch({
```
Comment (mikhail-angelov) [inline src/actions/chat.ts]: no need this

File: src/actions/chat.ts [hunk 2]
```diff
@@ -215,3 +218,53 @@ export const refreshChatList = () => async (dispatch) => {
...
+        data: RNFetchBlob.wrap(image.path),
+        type: image.mime
+      },
+    ]).uploadProgress({ interval: 50 }, (written, total) => {
+      dispatch({
+        type: UPLOAD_PROGRESS,
+        payload: written / total
+      });
+    })
+    .progress((received, total) => {
```
Comment (mikhail-angelov) [inline src/actions/chat.ts]: read documentation

File: src/components/Chat/Message.tsx [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
-export const Message = ({ text, idx, isMyMessage, timestamp }) => {
+const { width } = Dimensions.get('window')
+
+export const Message = ({ text, files, idx, isMyMessage, timestamp }) => {
 
   return (
     <MessageWrapper isMyMessage={isMyMessage}>
+      {_.map(files, fileUrl => (<Image
+        style={{
+          overflow: "hidden", width: width * 0.7, height: 180, borderWidth: 0.5, borderRadius: 10,
```
Comment (mikhail-angelov) [inline src/components/Chat/Message.tsx]: think how to make it 70%

Comment (mikhail-angelov) [reply src/components/Chat/Message.tsx]: use styled component

File: src/components/Chat/MessageInput.tsx [hunk 4]
```diff
@@ -54,26 +78,83 @@ export class MessageInput extends React.Component<IProps, IState> {
...
+          }}
           style={{ marginLeft: 12, marginRight: 12 }}
           size={22}
           name="paperclip"
           color={SOFT_BLUE_COLOR} />
+        <UploadPhotoWrap horizontal={true}>
+          {_.map(chat.imagesInCurrentMessage, imageUrl => (
+            <View>
+              <Image
+                style={{
```
Comment (mikhail-angelov) [inline src/components/Chat/MessageInput.tsx]: styled components

File: src/components/Chat/MessageInput.tsx [hunk 5]
```diff
@@ -54,26 +78,83 @@ export class MessageInput extends React.Component<IProps, IState> {
...
+                  marginBottom: 5, marginTop: 10, marginRight: 10, backgroundColor: `${WHITE_COLOR}`, borderColor: `${SOFT_BLUE_COLOR}`
+                }}
+                source={{ uri: imageUrl }}
+              />
+              <Icon
+                onPress={() => {
+                  this.props.deletePhotoInMessage(imageUrl)
+                }}
+                solid
+                style={{ overflow: "hidden", backgroundColor: `${WHITE_COLOR}`, borderRadius: 11, top: 0, right: 0, position: 'absolute' }}
```
Comment (mikhail-angelov) [inline src/components/Chat/MessageInput.tsx]: refactor

File: src/components/Chat/MessageInput.tsx [hunk 6]
```diff
@@ -54,26 +78,83 @@ export class MessageInput extends React.Component<IProps, IState> {
...
+                size={22}
+                name="times-circle"
+                backgroundColor={WHITE_COLOR}
+                color={SOFT_BLUE_COLOR} />
+            </View>
+          ))
+          }
+          {chat.uploadingPhoto && <View
+            style={{
```
Comment (mikhail-angelov) [inline src/components/Chat/MessageInput.tsx]: refactor

File: src/components/Chat/index.tsx [hunk 7]
```diff
@@ -32,7 +32,8 @@ class Chat extends React.PureComponent<IProps> {
   }
 
   public componentDidUpdate(prevProps) {
```
Comment (mikhail-angelov) [inline src/components/Chat/index.tsx]: remove it from component

File: src/actions/chat.ts (hunk 1 above)
Comment (KovalevAnton) [reply src/actions/chat.ts]: already did this in reducer 😄

**Comment by KovalevAnton** [comment]
@mikhail-angelov can i merge it?

File: src/actions/chat.ts [hunk 8]
```diff
@@ -184,34 +186,66 @@ export const changeChatPicture = (image, chat) => async (dispatch) => {
...
+        payload: written / total
       });
     })
-    .catch((err) => {
-      console.log(err)
-    })
+  dispatch({
+    type: UPLOAD_END
+  });
+  const newPicUrl = await JSON.parse(resp.data)
```
Comment (mikhail-angelov) [inline src/actions/chat.ts]: this is synchronous operation

File: src/components/Chat/MessagesList.tsx [hunk 9]
```diff
@@ -5,20 +5,37 @@ import styled from "styled-components";
...
+}
+export class MessagesList extends React.Component<IProps> {
+  public scrollToEnd = () => {
+    this.scrollView.scrollToEnd();
+  }
+  public render() {
+    const { messages, userEmail } = this.props
+    return (
+      <ScrollView
+        style={{
```
Comment (mikhail-angelov) [inline src/components/Chat/MessagesList.tsx]: should it be like styled component?

Comment (KovalevAnton) [reply src/components/Chat/MessagesList.tsx]: No, because ScrollView lose his context, and we can't use refs.

Comment (mikhail-angelov) [reply src/components/Chat/MessagesList.tsx]: ok

**Pull request merged**